apps/services/sessions/src/app/sessions/sessions.service.ts

```diff
@@ -11,11 +17,65 @@ export class SessionsService {
     private readonly sessionModel: typeof Session,
   ) {}
 
-  async findAll(user: User): Promise<Session[]> {
-    return []
+  async findAll(
+    user: User,
+    otherUser: string,
+    query: SessionsQueryDto,
+  ): Promise<SessionsResultDto> {
+    if (user.actor && !kennitala.isCompany(user.nationalId)) {
+      throw new ForbiddenException(
+        'Personal delegations are not allowed to get session data.',
+      )
+    }
+
+    let whereOptions: WhereOptions
+
+    if (user.actor) {
+      if (otherUser) {
+        whereOptions = {
+          actorNationalId: otherUser,
+          subjectNationalId: user.nationalId,
+        }
+      } else {
+        whereOptions = {
+          subjectNationalId: user.nationalId,
+        }
+      }
+    } else {
+      if (otherUser) {
+        whereOptions = {
+          [Op.or]: [
+            {
+              actorNationalId: user.nationalId,
+              subjectNationalId: otherUser,
+            },
+            { actorNationalId: otherUser, subjectNationalId: user.nationalId },
+          ],
+        }
+      } else {
+        whereOptions = {
+          [Op.or]: [
+            {
+              actorNationalId: user.nationalId,
+            },
+            { subjectNationalId: user.nationalId },
+          ],
+        }
+      }
+    }
+
+    return paginate({
+      Model: this.sessionModel,
+      limit: query.limit || 10,
```

Comment:
Wondering if we should have some upper limit on the user input, to prevent slow queries that could be harmful if the user tries to be funny and query million rows.

Reply:
Wonder if any user would have a million associated logins in a year.. Any suggestions what the limit should be?

Reply:
I would just start with 100. I don't think the UI is going to query for more than 100 at a time.

Reply:
Fixed.